recipe-server/normandy/recipes/models.py

```diff
@@ -517,19 +550,55 @@ def recipes_used_by(self):
     def __str__(self):
         return self.name
 
+    def canonical_json(self):
+        # Avoid circular import
+        from normandy.recipes.api.v1.serializers import ActionSerializer
+        data = ActionSerializer(self).data
+        return CanonicalJSONRenderer().render(data)
```

Comment:
We can use `normandy.base.utils.canonical_json_dumps` here instead of the renderer.